manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml

```
@@ -216,6 +216,9 @@ data:
     injectedAnnotations:
 
     template: |
+      {{- $holdProxyPC := boolValueOrDefault .ProxyConfig.HoldApplicationUntilProxyStarts "" }}
```

```suggestion
      {{- $holdProxyPC := .ProxyConfig.HoldApplicationUntilProxyStarts.GetValue }}
```

not certain it works, but i suspect it would

---

It doesn't. The problem here is that if you call GetValue it actually converts it to a boolean false if unset, and we need to preserve that information since the requested behavior is "If set in proxyconfig to either true or false, ignore the behavior of the meshconfig flag". Losing the 'nil' if it is unset would mean that the meshconfig global setting is ignored in all situations.